constraint/pkg/regorewriter/regorewriter_test.go

```diff
@@ -495,3 +506,369 @@ func TestRegoRewriterErrorCases(t *testing.T) {
+			prefix:  "foo.bar",
+			extern:  []string{"data.ext.util"},
+			imports: "data.ext",
+			want:    "data.foo.bar.ext",
+		},
+		{
+			name:    "add prefix for data ref",
+			prefix:  "foo.bar",
+			extern:  []string{"data.inventory"},
+			imports: "data.some_ref",
+			want:    "data.foo.bar.some_ref",
+		},
+		{
+			name:    "rewrite for ref same as libs",
+			prefix:  "foo.bar",
+			lib:     []string{"data.lib"},
+			imports: "data.lib",
+			want:    "data.foo.bar.lib",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			r, err := New(NewPackagePrefixer(tt.prefix), tt.lib, tt.extern)
+			if err != nil {
+				t.Fatalf("Failed to create RegoRewriter %v", err)
+			}
+			ref, err := ast.ParseRef(tt.imports)
+			if err != nil {
+				t.Fatalf("got ast.ParseRef(%q) error = %v, want nil", tt.imports, err)
+			}
+			wantRef, err := ast.ParseRef(tt.want)
+			if err != nil {
+				t.Fatalf("got ast.ParseRef(%q) error = %v, want nil", tt.want, err)
+			}
+			if got := r.rewriteDataRef(ref); !got.Equal(wantRef) {
+				t.Errorf("rewriteDataRef() = %q, want %q", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestAddEntryPointParseErrors(t *testing.T) {
+	tcs := []struct {
+		name      string
+		path      string
+		src       string
+		wantError error
+	}{
+		{
+			name: "add lib.rego path",
+			path: "data.stuff.foolib",
+			src: `package lib.rego
+
+test_ok {
+	true
+}
+`,
+		},
+		{
+			name: "add invalid rego",
+			path: "invalidrego",
+			src: `package lib.rego
+something invalid`,
+			wantError: ErrInvalidModule,
+		},
+	}
+
+	for _, tc := range tcs {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := MockRegoWriter()
+			if err != nil {
+				t.Fatalf("Failed to create RegoRewriter %q", err)
+			}
+			if gotErr := rr.AddEntryPoint(tc.path, tc.src); !errors.Is(gotErr, tc.wantError) {
+				t.Errorf("got AddEntryPoint() error = %q, want %v", gotErr, tc.wantError)
+			}
+		})
+	}
+}
+
+func TestRegoRewriterAddPathFromFs(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		wantErr error
+	}{
+		{
+			name:    "path not exist",
+			path:    "foo/bar",
+			wantErr: ErrReadingFile,
+		},
+		{
+			name: "path with test folder",
+			path: ".",
+		},
+		{
+			name: "path without test",
+			path: "../../",
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			rr, err := MockRegoWriter()
+			if err != nil {
+				t.Fatalf("Failed to create RegoRewriter %q", err)
+			}
+			if err := rr.AddLibFromFs(tc.path); !errors.Is(err, tc.wantErr) {
+				t.Errorf("AddLibFromFs() error = %v, wantErr %v", err, tc.wantErr)
+			}
+			if err := rr.AddBaseFromFs(tc.path); !errors.Is(err, tc.wantErr) {
+				t.Errorf("AddBaseFromFs() error = %v, wantErr %v", err, tc.wantErr)
+			}
+		})
+	}
+}
+
+func TestRegoRewriteEntryPoint(t *testing.T) {
+	tests := []struct {
+		name       string
+		prefix     string
+		content    string
+		wantResult map[string]string
+		wantError  error
```

Comment:
Is `wantError` used by any tests? Either there should be a test that exercises it, or it should be removed.

Reply:
removed